src/google-tag-manager.ts

```diff
@@ -83,8 +83,15 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
       this.logger.warn(`'${customEvent}' is an invalid player event`);
       return;
     }
+
     this.eventManager.listen(this.player, customEvent, (event: FakeEvent) => {
-      const dataLayerVariablePayload = event.payload !== undefined ? { [`${event.type}-payload`]: event.payload } : {};
+      const eventPayload = event.payload !== undefined ? event.payload : {};
+      if (typeof this.player.sources.metadata.entryId === 'string') {
```

Comment:
why does it matter if entryId is set ? and why does it guarantee that name will be set as well ?